.github/scripts/check-modified-subtree.sh

```diff
@@ -1,46 +1,123 @@
-    echo "::error::The MySQLite library has been edited. Please submit a PR to https://github.com/fptje/MySQLite instead!"
-    FAILED=true
-fi
 
-if echo "$DIFFED_FILES" | grep -qP "^gamemode/libraries/simplerr.lua"; then
-    echo "::error::The Simplerr library has been edited. Please submit a PR to https://github.com/fptje/simplerr instead!"
-    FAILED=true
-fi
+#   Template for the GitHub summary
 
-if echo "$DIFFED_FILES" | grep -qP "^gamemode/libraries/sh_cami.lua"; then
-    echo "::error::The CAMI library has been edited. Please submit a PR to https://github.com/glua/CAMI instead!"
-    FAILED=true
-fi
+template="
+    > [!WARNING]
+    > Files from **[{project}]({repository})** have been modified!  
+    > ➞ Changes should only be made in it's repository
+    > 
+    {files}
+    >
+"
 
-if echo "$DIFFED_FILES" | grep -qP "^gamemode/modules/fspectate/"; then
-    echo "::error::Files from FSpectate have been edited. Please submit a PR to https://github.com/fptje/FSpectate instead!"
-    FAILED=true
-fi
+#   Trim leading space for Markdown compatibility
+
+template="$( echo "$template" | sed 's/^[[:space:]]*//' )"
+
+
+failed=false
+
+function check {
+
+    local name="$1"
+    local path="^$2"
+    local repo="$3"
+
+    
+    #   Check if any matching files were modified
+    
+    local matched=$( 
+        echo "$files" |
+        grep --perl-regexp "$path"
+    )
+
+
+    #   Ignore if nothing matched
+
+    if [ -z "$matched" ] ; then
+        return
+    fi
+
+
+    #   Append info to GitHub's summary
+
+    local info="$template"
+
+    matched="$(printf -- '> `%s`  \n' "$matched")"
+    
+    info=${info//\{repository\}/$repo}
+    info=${info//\{project\}/$name}
+    info=${info//\{files\}/$matched}
 
-if [[ "$FAILED" = true ]]; then
+    echo "$info" >> $GITHUB_STEP_SUMMARY
+
+    failed=true
+}
+
+
+check   "Falco's Prop Protection"   \
+        'gamemode/modules/fpp/pp/'  \
+        'https://github.com/fptje/falcos-Prop-protection'
+
+
+check   "FN Library"                \
+        'gamemode/libraries/fn.lua' \
+        'https://github.com/fptje/GModFunctional'
+
+
+check   "MySQLite Library"                          \
+        'gamemode/libraries/mysqlite/mysqlite.lua'  \
+        'https://github.com/fptje/MySQLite'
+
+
+check   "Simplerr Library"                  \
+        'gamemode/libraries/simplerr.lua'   \
+        'https://github.com/fptje/simplerr'
+
+
+check   "CAMI Library"                      \
+        'gamemode/libraries/sh_cami.lua'    \
+        'https://github.com/glua/CAMI'
+
+
+check   "FSpectate"                     \
+        'gamemode/modules/fspectate/'   \
+        'https://github.com/fptje/FSpectate'
+
+
+if [[ "$failed" = true ]] ; then
+    echo 'Some dependencies have been modified!'
     exit 1
 fi
+
+
+echo 'No dependencies have been modified.'
+
+echo "**No dependencies have been modified**" >> $GITHUB_STEP_SUMMARY
```

Comment:
I prefer this check not posting any comments on success, as that would be clutter that requires some thinking to process. The Green checkmark should be good enough

Reply:
I removed the GitHub summary message~